probes/requiresLastPushApproval/impl.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//nolint:stylecheck
+package requiresLastPushApproval
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/branchprotection"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "requiresLastPushApproval"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	r := raw.BranchProtectionResults
+	var findings []finding.Finding
+
+	for i := range r.Branches {
+		branch := &r.Branches[i]
+		nilMsg := fmt.Sprintf("unable to retrieve whether 'last push approval' is required to merge on branch '%s'",
+			*branch.Name)
+		trueMsg := fmt.Sprintf("'last push approval' enabled on branch '%s'", *branch.Name)
+		falseMsg := fmt.Sprintf("'last push approval' disabled on branch '%s'", *branch.Name)
```

Comment:
more info / optional?

if the format for all these messages is similar, should we just accept setting name and branch name as params, and not nil, true, false messages? 
and then nil,true, and false can be autogenerated?
```
unable to retrieve whether %q is required to merge on branch %q
%q enabled on branch %q
%q disabled on branch %q
```